src/libstore/build.cc

```diff
@@ -2499,13 +2503,20 @@ void DerivationGoal::runChild()
             }
             sandboxProfile += ")\n";
 
+            sandboxProfile += additionalSandboxProfile;
+
             debug("Generated sandbox profile:");
             debug(sandboxProfile);
 
+            char *tmpProfile = strdup((format("%1%/nix-sandboxXXXXXX.sb") % globalTmpDir).str().c_str());
+            int profileFd = mkstemps(tmpProfile, 3);
+            closeOnExec(profileFd);
+            writeFull(profileFd, sandboxProfile);
```

Comment:
Any reason not to use https://github.com/pikajude/nix/blob/396b05ea9645b769326e2d55f908786ac905f12e/src/libutil/util.hh#L196-L206